src/App.tsx

```diff
@@ -1,61 +1,111 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/todo';
+import { getUserById } from './services/userService';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const initialTodo: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const [todos, setTodos] = useState<Todo[]>(initialTodo);
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    setHasTitleError(!trimmedTitle);
+    setHasUserIdError(!userId);
+
+    if (!trimmedTitle || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
```

Comment:
The calculation of the new `id` for a todo item using `Math.max(...todos.map(todo => todo.id)) + 1` can lead to errors if the `todos` array is empty. Consider handling the case where `todos` is empty to avoid `Math.max` returning `-Infinity`.